assets.src/src/redmine-tracky/controllers/timer-controller.ts

```diff
@@ -56,18 +55,26 @@ export default class extends Controller {
         }
     }
 
-    private timeDifference(): string {
-        const startDateTime = this.convertToDateTime((this.startTarget as HTMLInputElement).value);
-        const endDateTime = this.convertToDateTime((this.endTarget as HTMLInputElement).value);
+    private timeDiffToString(timeDiff: TimeDiff) {
+        const sign = (timeDiff.minutes < 0 || timeDiff.seconds < 0) ? '-' : '';
+        const [hours, mins, secs] = [timeDiff.hours, timeDiff.minutes, Math.round(timeDiff.seconds)]
+                                    .map((value) => value.toString().replace('-', '').padStart(2, '0'));
+        const hoursFormat = Number(hours) > 0 ? `${hours}:` : '';
+    
+        return `${sign}${hoursFormat}${mins}:${secs}`;
+    }
 
-        const duration: Duration = (endDateTime.isValid ? endDateTime : this.adjustedDateTime()).diff(
-            startDateTime.isValid ? startDateTime : this.adjustedDateTime(),
-            this.timeDiffFields
-        );
+    private dateTimeFromTarget(target: HTMLInputElement) {
+        const dateTime = this.convertToDateTime(target.value);
+        return dateTime.isValid ? dateTime : this.adjustedDateTime();
+    }
 
-        const timeDiff: TimeDiff = (duration as any).values as TimeDiff || {};
+    private timeDifference() {
+        const startDateTime = this.dateTimeFromTarget(this.startTarget);
+        const endDateTime = this.dateTimeFromTarget(this.endTarget);
+        const duration = endDateTime.diff(startDateTime, this.timeDiffFields);
 
-        return timeDiffToString(timeDiff);
+        return this.timeDiffToString((duration as any).values || {});
```

Comment:
why is the cast needed? i would pass in the duration, so it's easier to see if it's a negative value

```suggestion
        return this.timeDiffToString(duration);
```

Reply:
it should return `duration.values` not just `duration`